src/components/DateRangePickerInputController.jsx

```diff
@@ -130,14 +133,16 @@ export default class DateRangePickerInputController extends React.Component {
     const {
       startDate,
       isOutsideRange,
+      minimumNights,
       keepOpenOnDateSelect,
       onDatesChange,
     } = this.props;
 
     const endDate = toMomentObject(endDateString, this.getDisplayFormat());
 
     const isEndDateValid = endDate && !isOutsideRange(endDate) &&
-      !isInclusivelyBeforeDay(endDate, startDate);
+      (!isInclusivelyBeforeDay(endDate, startDate) ||
+      (minimumNights === 0 && isSameDay(endDate, startDate)));
```

Comment:
+1 to the first part, and ah yeah, I see how the `Inclusively` functions work now, should have taken a closer look -- `!isInclusivelyBeforeDay()` is equivalent to `isAfterDay()` so yeah, makes sense to me. Though we'd have to do `startDate.clone().add(minimumNights, 'days')`. Do you think it's worth the clone?

Thanks!